opensearch-operator/pkg/builders/cluster.go

```diff
@@ -1105,6 +1118,19 @@ func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *monitoring.ServiceMonito
 		},
 	}
 
+	if cr.Spec.General.Monitoring.Scheme == "" {
+		cr.Spec.General.Monitoring.Scheme = "https"
```

Comment:
Better not to overwrite the field in the spec but instead save it to a separate variable

Reply:
Removed this based off the comment below around not being able to manually set the scheme